src/components/pages/career-details/career-form/career-form.tsx

```diff
@@ -240,19 +258,20 @@ export const CareerForm: React.FC<CareerFormProps> = (props) => {
   if (isSubmitSuccessful) {
     return <SuccessMessage />;
   }
+  const onSubmit = handleSubmit(onSubmitHandler, onErrorHandler);
 
   return (
-    <CareerFormStyled onSubmit={handleSubmit(onSubmit, onError)}>
+    <CareerFormStyled onSubmit={onSubmit}>
       <Headline>{t('career.headline')}</Headline>
       <Fieldset disabled={isSubmitting}>
         <Grid nested>
           <CareerTextFields register={register} errors={errors} />
 
           {/*File-Upload*/}
           <FileUpload
+            name={'documents'}
```

Comment:
I can't tell any difference between the old and the new way of passing this property. In addition, you hardcoded the value in the `FileUploadProps` interface... do we actually need to pass this at all? One of the places smells.

Reply:
Not sure I understand this correctly. Does it still make a difference if you use the "unbound" way (`name="documents")`? Either way, fine with me. Thanks for the explanation 👌

Reply:
reverted the assignment.

@fischeversenker yes it smells a lot but it's the most pragmatic while functional approach to adhere to the typing given by the library. The `register` functions etc are all typed. This means if you create a sub components you have to relate that component somehow to the parent type (here the `CareerFormValue`).

I'm glad I found a way that is minimal invasive by pinning the type, hence the following change you mentioned:

```
interface FileUploadProps {
  name: 'documents';
```

A `name: string` will not work as the derived functions `register`, `setError` etc. all refer to the parent `CareerFormValue` type and an untyped name will not work.

I consider the entire form a technical debt. That's why I decided not to spend more time than useful on this topic. I hope this is okay for you, otherwise I'm happy to hear how you would approach this besides refactoring all involved components 🙏